fe/fe-core/src/main/java/com/starrocks/binlog/BinlogManager.java

```diff
@@ -0,0 +1,180 @@
+        }
+        binlogLock.readLock().unlock();
+        return false;
+    }
+
+    // the caller hold the write lock of Database
+    public void checkAndSetBinlogAvailableVersion(Long toPublishTxnId, Set<Long> runningTransactionId,
+                                                 Database db) {
+        binlogLock.readLock().lock();
+        for (Long binlogTxnId : transcationidToBinlogTxnId.get(toPublishTxnId)) {
+            boolean isLast = true;
+            for (Long transactionId : runningTransactionId) {
+                if (transactionId != toPublishTxnId && transactionId < binlogTxnId) {
+                    isLast = false;
+                }
+            }
+            if (isLast) {
+                List<Long> tableIds = binlogTxnIdToTableIds.get(binlogTxnId);
+                for (Long tableId : tableIds) {
+                    if (db.getTable(tableId).isOlapTable()) {
+                        OlapTable olapTable = (OlapTable) db.getTable(tableId);
+                        Collection<Partition> partitions =  olapTable.getPartitions();
+                        for (Partition partition : partitions) {
+                            // TODO persist the binlogAvailableVersion to ediglog
+                            partition.setBinlogAvailableVersion(partition.getVisibleVersion() + 1);
+
+                        }
+                    }
+
+                }
+            }
+        }
+        binlogLock.readLock().lock();
+    }
+
+    // the caller don't need to hold the db lock
+    public boolean tryEnableBinlog(Database db, Long tableId, Long binlogTtL, Long binlogMaxSize) {
+        BinlogContext context = new BinlogContext(); // 修改FE元数据成功后，会将其中标志置为true
+        // pass properties not binlogConfig is for
+        // unify the logic of alter table manaul and alter table of mv trigger
+        HashMap<String, String> properties = new HashMap<>();
+        properties.put(PropertyAnalyzer.PROPERTIES_ENABLE_BINLOG, "true");
+        properties.put(PropertyAnalyzer.PROPERTIES_BINLOG_TTL, String.valueOf(binlogTtL));
+        properties.put(PropertyAnalyzer.PROPERTIES_BINLOG_MAX_SIZE, String.valueOf(binlogMaxSize));
+        try {
+            schemaChangeHandler.updateBinlogConfigMeta(db, tableId, properties, context);
+        } catch (DdlException exception) {
+            return context.isModifyFeMetaSuccess();
+        }
+        return true;
+
+    }
+
+    // the caller don't need to hold the db lock
+    public boolean tryCloseBinlog(Database db, Long tableId, Long dbId) {
+        try {
+            // modify FE meta, return true
+            // try to distrucite BE task best effort
+            HashMap<String, String> properties = new HashMap<>();
+            properties.put(PropertyAnalyzer.PROPERTIES_ENABLE_BINLOG, "false");
+            schemaChangeHandler.updateTableMeta(db, db.getTable(tableId).getName(), properties, TTabletMetaType.DISABLE_BINLOG);
+        } catch (DdlException e) {
+            return false;
```

Comment:
Log the exception ?

Reply:
in fact, no exception will be thrown when disable binlog, it is just sharing a method，updateTableMeta，which may throw exception when enable binlog